src/plugins/hiddenColumns/contextMenuItem/showColumn.js

```diff
@@ -2,112 +2,83 @@ import { rangeEach } from '../../../helpers/number';
+  const hiddenBefore = [];
+  const hiddenBetween = [];
+  const hiddenAfter = [];
 
   return {
     key: 'hidden_columns_show',
     name() {
-      const selection = this.getSelectedLast();
-      let pluralForm = 0;
-
-      if (Array.isArray(selection)) {
-        let [, fromColumn, , toColumn] = selection;
-
-        if (fromColumn > toColumn) {
-          [fromColumn, toColumn] = [toColumn, fromColumn];
-        }
-
-        let hiddenColumns = 0;
-
-        if (fromColumn === toColumn) {
-          hiddenColumns = beforeHiddenColumns.length + afterHiddenColumns.length;
-
-        } else {
-          rangeEach(fromColumn, toColumn, (column) => {
-            if (hiddenColumnsPlugin.isHidden(column)) {
-              hiddenColumns += 1;
-            }
-          });
-        }
-
-        pluralForm = hiddenColumns <= 1 ? 0 : 1;
-      }
+      const hiddenColumnsLength = hiddenBefore.length + hiddenBetween.length + hiddenAfter.length;
+      const pluralForm = hiddenColumnsLength <= 1 ? 0 : 1;
 
       return this.getTranslatedPhrase(C.CONTEXTMENU_ITEMS_SHOW_COLUMN, pluralForm);
     },
     callback() {
-      const { from, to } = this.getSelectedRangeLast();
-      const start = Math.min(from.col, to.col);
-      const end = Math.max(from.col, to.col);
+      if (hiddenBetween.length) {
+        hiddenColumnsPlugin.showColumns(hiddenBetween);
 
-      if (start === end) {
-        if (beforeHiddenColumns.length === start) {
-          hiddenColumnsPlugin.showColumns(beforeHiddenColumns);
-          beforeHiddenColumns.length = 0;
+      } else {
+        if (hiddenBefore.length) {
+          hiddenColumnsPlugin.showColumns(hiddenBefore);
         }
-        if (afterHiddenColumns.length === this.countSourceCols() - (start + 1)) {
-          hiddenColumnsPlugin.showColumns(afterHiddenColumns);
-          afterHiddenColumns.length = 0;
+        if (hiddenAfter.length) {
+          hiddenColumnsPlugin.showColumns(hiddenAfter);
         }
-
-      } else {
-        rangeEach(start, end, column => hiddenColumnsPlugin.showColumn(column));
       }
 
       this.render();
       this.view.wt.wtOverlays.adjustElementsSize(true);
     },
     disabled: false,
     hidden() {
-      if (!hiddenColumnsPlugin.hiddenColumns.length || !this.selection.isSelectedByColumnHeader()) {
+      if (!this.selection.isSelectedByColumnHeader() || hiddenColumnsPlugin.getHiddenColumns().length < 1) {
         return true;
       }
 
-      beforeHiddenColumns.length = 0;
-      afterHiddenColumns.length = 0;
+      hiddenBefore.length = 0;
+      hiddenBetween.length = 0;
+      hiddenAfter.length = 0;
 
-      const { from, to } = this.getSelectedRangeLast();
-      const start = Math.min(from.col, to.col);
-      const end = Math.max(from.col, to.col);
-      let hiddenInSelection = false;
+      let shouldBeHidden = true;
+      let [, start, , end] = this.getSelectedLast();
+      [start, end] = [Math.min(start, end), Math.max(start, end)];
```

Comment:
Isn't that an unnecessary array allocation?
```suggestion
      start = Math.min(start, end);
```
```suggestion
      end = Math.max(start, end);
```